clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4707,6 +4711,36 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
     O << "  __SYCL_DLL_LOCAL\n";
     O << "  static constexpr bool isESIMD() { return " << K.IsESIMDKernel
       << "; }\n";
+    O << "  __SYCL_DLL_LOCAL\n";
+    O << "  static constexpr const char* getFileName() { return \""
+#ifndef NDEBUG
```

Comment:
Why is what gets filled in here based on the 'debug' status of the COMPILER, not the debug state of the COMPILATION?  The former is basically useless.

Reply:
https://github.com/intel/llvm/pull/5335/commits/16d6af3670f802e0ec44f662baaf6aed5d1f370a